geozero/src/wkt/wkt_writer.rs

```diff
@@ -53,23 +57,26 @@ impl<'a, W: Write> WktWriter<'a, W> {
         }
         Ok(())
     }
-    fn geom_begin(&mut self, idx: usize, tag: &[u8]) -> Result<()> {
+    fn geom_begin(&mut self, tag: &[u8], tagged: bool, size: usize, idx: usize) -> Result<()> {
         self.header(self.srid)?;
-        self.comma(idx)?;
-        self.out.write_all(tag)?;
-        Ok(())
-    }
-    fn tagged_geom_begin(&mut self, tagged: bool, idx: usize, tag: &[u8]) -> Result<()> {
         self.comma(idx)?;
         if tagged {
             self.out.write_all(tag)?;
+        }
+        self.geometry_sizes.push(size);
+        if size == 0 {
+            self.out.write_all(b" EMPTY")?;
         } else {
             self.out.write_all(b"(")?;
         }
         Ok(())
     }
     fn geom_end(&mut self) -> Result<()> {
-        self.out.write_all(b")")?;
+        if let Some(geometry_size) = self.geometry_sizes.pop() {
```

Comment:
Would we ever expect this to be `None`?

Reply:
A geometry would have to end before it began, so in theory no ?

Reply:
Consider adding something like `else { debug_assert!(false, "ended geometry that didn't start") }` as a "this should never happen but we don't want to blow up in production if it does"

Reply:
Done 👍

Reply:
Nice catch... https://github.com/georust/geozero/pull/160#issuecomment-1676431784